cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -50,89 +54,110 @@ struct has_negative_nans {
 };
 
 /**
- * @brief A structure to be used along with type_dispatcher to check if a
- * `column_view` has any negative NaN entry
+ * @brief A structure to be used along with type_dispatcher to check if a column has any
+ * negative NaN value.
  */
-struct has_negative_nans_fn {
+struct has_negative_nans_dispatch {
   template <typename Type, std::enable_if_t<cuda::std::is_floating_point_v<Type>>* = nullptr>
   bool operator()(column_view const& lists_entries, rmm::cuda_stream_view stream) const noexcept
   {
     auto const d_entries = column_device_view::create(lists_entries, stream);
-    return thrust::count_if(rmm::exec_policy(stream),
-                            thrust::make_counting_iterator(0),
-                            thrust::make_counting_iterator(lists_entries.size()),
-                            detail::has_negative_nans<Type>{*d_entries, lists_entries.has_nulls()});
+    return thrust::count_if(
+      rmm::exec_policy(stream),
+      thrust::make_counting_iterator(0),
+      thrust::make_counting_iterator(lists_entries.size()),
+      detail::has_negative_nans_fn<Type>{*d_entries, lists_entries.has_nulls()});
+  }
+
+  template <typename Type, std::enable_if_t<std::is_same_v<Type, cudf::struct_view>>* = nullptr>
```

Comment:
I'm not entirely sure, is this really noexcept? There are a lot of function calls here (thrust and libcudf) and I'm not confident that all of them are marked that way or whether we might trigger UB here. Same goes for all the associated functors.

Reply:
Thanks. Since this func involves recursive calls, it may throw. I fixed it.